src/libutil/source-accessor.hh

```diff
@@ -176,6 +176,39 @@ struct SourceAccessor : std::enable_shared_from_this<SourceAccessor>
         const CanonPath & path,
         SymlinkResolution mode = SymlinkResolution::Full);
 
+    /**
+     * When `resolveSymlinks` encounters an absolute target, that path
+     * may or may not denote a local inside the current source accessor
+     * --- there is no general answer to the question, it depends on
+     *  what the current source accessor "represents".
+     *
+     * For now, we use this variable to decide: if the path has this
+     * path as a prefix, then the suffix points inside this store
+     * prefix. If the path does not have path as prefix, than the
+     * symlink escapes the source acccessor.
```

Comment:
I'm not really following this description. In "if the path", what path does "the path" refer to? The symlink target? And what does "store prefix" mean in the general context of accessors? And what's the result of the decision (i.e. what happens if the symlink does / does not escape)?

Reply:
Probably best to give a concrete example here, e.g. a symlink target `/nix/store/foo` should resolve to `/foo` for accessors whose root is considered to be `/nix/store`.

Reply:
OK added example